### DISCUSSION TITLE
Validation support

### DISCUSSION THREAD

**Discussion by eandre** [encoredev/encore#330] [Suggestions]
We're looking at adding better support for validating incoming requests. While we intend on providing a fully-fledged validation library in the future (and we have early designs for that that are looking promising), we also want to make it easy to do request validation today.

To make that happen, we propose defining an interface: `interface { Validate() error }` that, if implemented by a request payload, will result in Encore automatically calling the `Validate` method after decoding the request. If the `Validate` method reports a non-nil error the request will be aborted and the client will receive a validation error.

If the validation function returns an [`*errs.Error`](https://encore.dev/docs/develop/errors) that error is reported unmodified to the caller. Other errors are converted to an `*errs.Error` with code `InvalidArgument`, which results in a HTTP response with status code `400 Bad Request`.

This design means that it's easy to use your validation library of choice. In the future we're looking to provide an out-of-the-box validation library for an even better developer experience.

Let us know what you think!

**Comment by marcuskohlberg** [comment]
For visibility, this has been implemented (a long time ago!)
